### ISSUE TITLE
console should support evaluating javascript expressions

### ISSUE THREAD

**Issue by skybrian** [sdbg/sdbg#101]
Unless I missed it, it looks like the sdbg console only supports displaying output written to the console. It would be useful to also be able to evaluate expression, similar to the console in a web browser.

In the remote debugging protocol, the commands are Runtime.evaluate [1] and possibly evaluateOnCallFrame [2] if you're stopped at a breakpoint.

[1] https://developer.chrome.com/devtools/docs/protocol/tot/runtime#command-evaluate

[2] https://developer.chrome.com/devtools/docs/protocol/tot/debugger#command-evaluateOnCallFrame

**Comment by ivmarkov** [comment]
Use the Eclipse "Expressions" view:
![expressions](https://cloud.githubusercontent.com/assets/2607589/5773601/f3ae4942-9d6c-11e4-9706-be81b3a9e0a2.png)

**Comment by skybrian** [comment]
Watched variables are useful but not the same, actually, since they don't work for expressions with side effects. For example, if I evaluate element.appendChild(), it shouldn't be reevaluated on each step when stepping through code.

Also, evaluating an expression at the console is useful in a different way because it keeps a history of the previous values of the expression and this is interleaved with log statements.

**Comment by ivmarkov** [comment]
I think what you want then is the Display view in Eclipse. However, it works only for Java code. I think we need a custom Display view which works for JavaScript code then. I'll reopen this issue and reformulate accordingly

**Comment by ivmarkov** [comment]
On a second thought, it can also be useful indeed if the Eclipse SDBG Console is extended to a real REPL display....
I guess I'm just too used to the Eclipse Java way, where you write your expression in the Display view and then if they have printing side effects, you get them on the Java console, which is a separate view...

**Comment by skybrian** [comment]
Yes, JavaScript developers are used to having a real REPL in their debugger. I think sdbg should aim to be a good Java debugger and a good JavaScript debugger, since GWT programs contain both Java and JavaScript.
